### PR TITLE
Deposit accept many

### PR DESCRIPTION
**Pull request by setzeus**
## Description
This PR introduces a 'complete-deposits-wrapper' that address the ability for signers to submit multiple deposits requests at once.

The motivation for this PR was to close out #86.

## Type of Change
This is a new feature change that wraps around existing functionality.

## Does this introduce a breaking change?
No.

## Are documentation updates required?
No.

## Testing information
Changes are tested using ClarinetSDK.

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `pnpm test` passes
- [ ] Changelog is updated
- [x] Tag 1 of @person1 or @person2

### CODE REVIEW

**Comment by radicleart** [comment]
@setzeus  - It looks like `complete-deposit-wrapper` can be called by the contract deployer - up until the `current-signer-principal` is changed in the sbtc-registry. Are there any potential unintentional possible effects of this ? Is there another guard?

**Comment by radicleart** [comment]
The sbtc-deposits contract does not need to verify the transactions using clarity-bitcoin - is this because calls can only be initiated by the node and the bitcoin transactions are already confirmed?

**Comment by setzeus** [comment]
> @setzeus - It looks like `complete-deposit-wrapper` can be called by the contract deployer - up until the `current-signer-principal` is changed in the sbtc-registry. Are there any potential unintentional possible effects of this ?

Temporary, for testing purposes.

**Comment by setzeus** [comment]
> The sbtc-deposits contract does not need to verify the transactions using clarity-bitcoin - is this because calls can only be initiated by the node and the bitcoin transactions are already confirmed?

Correct.

**Review by radicleart — APPROVED** [review]
Thanks for answering queries - looks good to me !

File: contracts/contracts/sbtc-deposit.clar [hunk 1]
```diff
@@ -7,12 +7,14 @@
 (define-constant dust-limit u546)
```
[the remark this answers is not in the document]
Comment (setzeus) [reply contracts/contracts/sbtc-deposit.clar:7]: That's a good Q. Going to open up an issue for a discussion on changing this to a variable & creating a setter/getter.

File: contracts/contracts/sbtc-deposit.clar [hunk 2]
```diff
@@ -28,11 +30,12 @@
...
             (replay-fetch (contract-call? .sbtc-registry get-completed-deposit txid vout-index))    
         )
 
-        ;; TODO
-        ;; Check that tx-sender is the bootstrap signer
+        ;; Check that the caller is the current signer principal
+        (asserts! (is-eq (get current-signer-principal current-signer-data) tx-sender) ERR_INVALID_CALLER)
 
         ;; Check that amount is greater than dust limit
         (asserts! (> amount dust-limit) ERR_LOWER_THAN_DUST)
```
[the remark this answers is not in the document]
Comment (setzeus) [reply contracts/contracts/sbtc-deposit.clar:41]: Copy. Fixing.

File: contracts/contracts/sbtc-deposit.clar [hunk 3]
```diff
@@ -56,11 +59,33 @@
...
+        (fold complete-individual-deposits-helper deposits (ok u0))
+    )
+)
 
 ;; read only functions
 ;;
 
 ;; private functions
-;;
+(define-private (complete-individual-deposits-helper (deposit {txid: (buff 32), vout-index: uint, amount: uint, recipient: principal}) (helper-response (response uint uint)))
```
[the remark this answers is not in the document]
Comment (setzeus) [reply contracts/contracts/sbtc-deposit.clar:80]: Not entirely sure either...will do.

**Pull request merged**